modules/search/instant-search/components/api.js

```diff
@@ -3,26 +3,73 @@
  */
 import fetch from 'unfetch';
 
-const FIELDS = [
-	'author',
-	'comment_count',
-	'date',
-	'excerpt_html',
-	'gravatar_url',
-	'permalink.url.raw',
-	'title_html',
-];
+/**
+ * @preserve jquery-param (c) 2015 KNOWLEDGECODE | MIT
+ * @param {object} a - parameters to add
+ * @returns {string} url query string
+ * From https://github.com/knowledgecode/jquery-param
+ */
+function query_params( a ) {
+	//eslint-disable-next-line no-var
+	var s = [];
+	const add = function( k, v ) {
+		v = typeof v === 'function' ? v() : v;
+		//eslint-disable-next-line no-nested-ternary
+		v = v === null ? '' : v === undefined ? '' : v;
+		s[ s.length ] = encodeURIComponent( k ) + '=' + encodeURIComponent( v );
+	};
+	const buildParams = function( prefix, obj ) {
+		let i, len, key;
 
-function stringifyArray( fieldName, array ) {
-	return array.map( ( element, index ) => `${ fieldName }[${ index }]=${ element }` ).join( '&' );
+		if ( prefix ) {
+			if ( Array.isArray( obj ) ) {
+				for ( i = 0, len = obj.length; i < len; i++ ) {
+					buildParams(
+						prefix + '[' + ( typeof obj[ i ] === 'object' && obj[ i ] ? i : '' ) + ']',
+						obj[ i ]
+					);
+				}
+			} else if ( String( obj ) === '[object Object]' ) {
+				for ( key in obj ) {
+					buildParams( prefix + '[' + key + ']', obj[ key ] );
+				}
+			} else {
+				add( prefix, obj );
+			}
+		} else if ( Array.isArray( obj ) ) {
+			for ( i = 0, len = obj.length; i < len; i++ ) {
+				add( obj[ i ].name, obj[ i ].value );
+			}
+		} else {
+			for ( key in obj ) {
+				buildParams( key, obj[ key ] );
+			}
+		}
+		return s;
+	};
+	return buildParams( '', a ).join( '&' );
 }
 
-function getAPIUrl( siteId, query ) {
-	return `https://public-api.wordpress.com/rest/v1.3/sites/${ siteId }/search?query=${ encodeURIComponent(
-		query
-	) }&${ stringifyArray( 'fields', FIELDS ) }`;
-}
+export function search( siteId, query, aggs, filter, resultFormat ) {
+	let fields = [];
+	let highlight_fields = [];
+	switch ( resultFormat ) {
+		case 'engagement':
+		case 'product':
+		case 'minimal':
+		default:
+			highlight_fields = [ 'title', 'content', 'comments' ];
+			fields = [ 'date', 'permalink.url.raw', 'tag.name.default', 'category.name.default' ];
+	}
 
-export function search( siteId, query ) {
-	return fetch( getAPIUrl( siteId, query ) );
+	const obj = {
+		query: query,
+		fields: fields,
+		highlight_fields: highlight_fields,
+		aggregations: aggs,
+		filter: filter,
+	};
+	return fetch(
+		`https://public-api.wordpress.com/rest/v1.3/sites/${ siteId }/search?` + query_params( obj )
```

Comment:
Same as above; since we're using es6 template strings for the first part of the URL, can we use them for everything?

```suggestion
		`https://public-api.wordpress.com/rest/v1.3/sites/${ siteId }/search?${ query_params( obj ) }`
```